src/main.ts

```diff
@@ -13,18 +13,56 @@ export async function run(): Promise<void> {
     const secrets: string[] = core.getMultilineInput("secrets", {
       required: true,
     });
+    const cloudRegion = core.getInput("cloud_region");
     const baseUrl: string = core.getInput("base_url");
     let identityUrl: string = core.getInput("identity_url");
     let apiUrl: string = core.getInput("api_url");
 
     core.info("Validating bitwarden/sm-action inputs...");
-    if (baseUrl) {
-      if (!isValidUrl(baseUrl)) {
-        throw TypeError("input provided for base_url not in expected format");
+
+    let customUrls;
+    //Check if only identityUrl is set and not apiUrl: if so throw an error, otherwise mark this as the user is using customUrls
+    if (identityUrl || apiUrl) {
+      //If either are set, make sure both are.
+      if (!identityUrl || !apiUrl) {
+        throw TypeError("If using custom Urls, Both identityUrl and apiUrl need to be set.");
       }
-      identityUrl = baseUrl + "/identity";
-      apiUrl = baseUrl + "/api";
+      customUrls = true;
     }
+    const selfHosted = baseUrl || customUrls;
+    if (selfHosted) {
+      //If self-hosted cloudBaseUrl will have no impact, but customUrls (apiUrl and IdentityUrl) will have an impact.
```

Comment:
I know this came from @cd-bitwarden suggestion, but i don't fully understand this comment. Maybe we remove it, since the code is self simple to read ?